### PR TITLE
[server] Reduce messagebus load (drastically)

### LINKED ISSUE #6662: GitpodMetaMessagebusTotalQueues firing too often
Investigate if either:
 - the query is wrong
 - the limit is too conservative
 - we have a real problem that requires fixing

### PR DESCRIPTION
**Pull request by geropl**
## Description
This fixes our current issues with "too many messages queues".

After investigating potential leaks in `server` there were no obvious ones. But it turns out that just create a ton of websocket connections. And each websocket connection creates `n` queues (where `n` [increased substantially lately with the introduction of T&P](https://github.com/gitpod-io/gitpod/blob/d4ecb355aada111d91e12c6b13a8470e65c37bf2/components/server/ee/src/workspace/gitpod-server-impl.ts#L85-L93)) ([details](https://github.com/gitpod-io/gitpod/blob/d4ecb355aada111d91e12c6b13a8470e65c37bf2/components/server/src/messaging/local-message-broker.ts#L43-L56)). 

The [pattern we see in production](https://github.com/gitpod-io/gitpod/issues/6662#issuecomment-968858687) regularly is thus on "us leaking connections/queues", but more "users leaking tabs". :grimacing: 
While we should also look into reducing the number of websocket connections from dashboard, there are some we cannot reduce (2 per workspace, for instance). Instead, this PR attempts to reduce the factor of "queues per connection" from `n` to `0` by:
 - establishing only one messagebus listener per server instance per topic (listening to _all_ messages)
 - doing the message [filtering and delivery in `server`](https://github.com/gitpod-io/gitpod/blob/d4ecb355aada111d91e12c6b13a8470e65c37bf2/components/server/src/messaging/local-message-broker.ts#L72-L126) (turns out to be simple)

This has the benefits that:
 - it clearly shows how much of the rabbitmq features we use (none)
 - makes it easier to swap out rabbitmq with something else
 - is a "drop-in" replacement for the current code, so we do not have to verify every dashboard page to still receive the correct messages (nr and order of messages stays identical)

Credits to @AlexTugarev for the idea and discussions! :pray: 

## Related Issue(s)

Fixes #6662.

## How to test

 1. start a workspace, open & close ports, and stop it
 2. notice how the dashboards updates as expected

## Release Notes

```release-note
NONE
```

## Documentation

- [x] /werft with-clean-slate-deployment

### CODE REVIEW

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please ask for approval from **geropl** after the PR has been reviewed.

*No associated issue*. Update pull-request body to add a reference to an issue, or get approval with `/approve no-issue`

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[components/gitpod-messagebus/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/gitpod-messagebus/OWNERS)**
- **[components/server/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/server/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: components/gitpod-messagebus/src/messagebus.ts [hunk 1]
```diff
@@ -7,14 +7,15 @@
 import { connect, Channel, Connection, Options, ConfirmChannel, Message } from "amqplib";
 import { injectable, inject } from 'inversify';
 import { Disposable } from "@gitpod/gitpod-protocol";
-export type WorkspaceSubtopic = "updates" | "log" | "credit" | "headless-log" | "ports";
-export const MessageBusHelper = Symbol("MessageBusHelper");
```
Comment (iamtangram) [inline components/gitpod-messagebus/src/messagebus.ts:11]: '''

**Comment by AlexTugarev** [comment]
/lgtm

tested with workspaces list and with prebuilds list, it worked as expected. 

looking forward to test ride this on gitpod-staging.com. 👍🏻

**Comment by roboquat** [comment]
LGTM label has been added.  <details>Git tree hash: 261236544c632bbd13643e68132ffabc44c22d75</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/6748#issuecomment-971715280" title="LGTM">AlexTugarev</a>*

Associated issue: *#6662*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[components/gitpod-messagebus/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/gitpod-messagebus/OWNERS)~~ [AlexTugarev]
- ~~[components/server/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/server/OWNERS)~~ [AlexTugarev]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**